lib/generator/ephemeral_data.dart

```diff
@@ -127,7 +128,7 @@ class Context {
         currentType: nextType,
         currentFieldName: nextFieldName,
         currentClassName: nextClassName,
-        ofUnion: ofUnion ?? this.ofUnion,
+        ofUnion: ofUnion == null ? this.ofUnion : ofUnion.value,
```

Comment:
As I've commented via DM, maybe we should (but not now 😂) to consider migrating `Context` to freezed or something similar ([ref](https://pub.dev/packages/freezed#copywith)).

Reply:
I've got a positive feedback https://github.com/comigor/artemis/issues/284#issuecomment-831127476

I would prefer to merge this PR. The migration of `context` to `freezed` would add an additional dependancy and will not make `everything great again`.
I looked through the `copy` methods in `Context` and it seems to me that some of them could be merged.

Reply:
Sure, let's merge this!